crates/libcontainer/src/container/builder_impl.rs

```diff
@@ -147,7 +147,7 @@ impl ContainerBuilderImpl {
         // This container_args will be passed to the container processes,
         // therefore we will have to move all the variable by value. Since self
         // is a shared reference, we have to clone these variables here.
-        let container_args = ContainerArgs {
+        let mut container_args = ContainerArgs {
```

Comment:
I prefer that `container_args` be immutable. Do we really need `mut`?

Reply:
I apologize for the delayed response.

I removed `mut` from `container_args`.
Since I need to pass the `container` inside `container_args` to run_hooks, I decided to call clone() at that point.

```rust
if let Some(mut container_for_hooks) = container_args.container.clone() {
    container_for_hooks.set_pid(init_pid.as_raw());
}
```

The reason for cloning is that the `container` at that moment doesn’t have its PID set, so I need a mutable clone to set the PID before passing it to the hook.